corehq/apps/users/views/mobile/users.py

```diff
@@ -1126,6 +1128,71 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
+class DeleteCommCareUsers(BaseManageCommCareUserView):
+    urlname = 'delete_commcare_users'
+    page_title = ugettext_noop('Bulk Delete')
+    template_name = 'users/bulk_delete.html'
+
+    @property
+    def page_context(self):
+        context = self.main_context
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(),
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
```

Comment:
there is a lot happening happening in this method which makes it a little hard to follow and possibly brittle

How about either
1. breaking it into methods
2. making a utility class that handles this request.

I would vote for utility class, @esoergel i know would vote for methods because he likes methods :)

Reply:
I like _functions_ :grin: (or staticmethods, which are just namespaced functions).  Methods can be nice too, when not complicated by inheritance or mutating instance variables.  Methods are sometimes necessary where there are a lot of flag arguments or memoization.

Reply:
Sorry, functions :)

Reply:
broke this up into smaller functions in 36d3198235c